File: Sample/MaaSAdvertising-iOS-Sample/InterstitialViewController.m
//
//  InterstitialViewController.m
//  PWAdvertising
//
//  Created on 4/16/16.
//  Copyright © 2016 Phunware, Inc. All rights reserved.
//

#import "InterstitialViewController.h"
#import <PWAdvertising/PWAdsInterstitial.h>
#import <PWAdvertising/PWAdsRequest.h>

// Example zone id, don't use this in your app!
static NSString * const kZoneID =  @"7271";

@interface InterstitialViewController () <PWAdsInterstitialDelegate>

@end

@implementation InterstitialViewController {
    PWAdsInterstitial *_interstitial;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // Create PWAdsInterstitial instance
    _interstitial = [PWAdsInterstitial new];
    _interstitial.delegate = self;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)loadButtonPressed:(id)sender {
    // Create ad request with the given zone id.
    PWAdsRequest *adsRequest = [PWAdsRequest requestWithZoneID:kZoneID];
    
    // Set test mode for request. This will fetch test ads.
    adsRequest.testMode = YES;
    
    // Load the ad created request from the interstitial view.
    [_interstitial loadAdsRequest:adsRequest];
}

#pragma mark -  PWAdsInterstitialDelegate

//  Called when a new advertisement is loaded.
- (void)interstitialDidLoadAd:(PWAdsInterstitial *)interstitialAd {
    NSLog(@"interstitialDidLoadAd:");
    
    // Interstitial ad is loaded. Now present the interstitial.
    [_interstitial presentFromViewController:self];
}

//  Called when an interstitial fails to load advertisement.
- (void)interstitial:(PWAdsInterstitial *)interstitial didFailWithError:(NSError *)error {
    NSLog(@"interstitial:didFailWithError:");
}

// Called before interstitial ad is dissmissed.
- (void)interstitialWillDismissModal:(PWAdsInterstitial *)interstitialAd {
    NSLog(@"interstitialWillDismissModal:");
}

// Called after Interstitial ad has been dismissed.
- (void)interstitialDidDismissModal:(PWAdsInterstitial *)interstitialAd {
    NSLog(@"interstitialDidDismissModal:");
}

// Called before user leaves the application. This happens when user taps on an advertisment.
- (BOOL)shouldLeaveApplicationForInterstitial:(PWAdsInterstitial *)interstitial {
    NSLog(@"shouldLeaveApplicationForInterstitial:");
    return YES;
}

@end

File: Samples/Sample/Sample/InterstitialViewController.swift
//
//  InterstitialViewController.swift
//  Sample
//
//  Created on 8/15/17.
//  Copyright © 2017 Phunware, Inc. All rights reserved.
//

import UIKit
import PWAdvertising

struct InterstitialConstants {
    static let iPhoneZoneId = "7271"
    static let iPadZoneId = "33761"
}

class InterstitialViewController: UIViewController {
    
    // MARK: - Properties
    var zoneId: String!
    var request = PWAdsRequest()
    var interstitial = PWAdsInterstitial()
    
    // MARK: - IBActions
    @IBAction func load(_ sender: UIButton) {
        request.testMode = true
        print("Zone Id: \(request.zoneID!)\nTest mode is \(request.testMode)\n")
        interstitial.load(request)
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        interstitial.delegate = self
        
        switch UIDevice.current.userInterfaceIdiom {
        case .pad:
            zoneId = InterstitialConstants.iPadZoneId
        default:
            zoneId = InterstitialConstants.iPhoneZoneId
        }
        request.zoneID = zoneId
    }
}

extension InterstitialViewController: PWAdsInterstitialDelegate {
    func interstitialDidLoadAd(_ interstitialAd: PWAdsInterstitial!) {
        print("interstitialDidLoadAd\n")
        interstitialAd.present(from: self)
    }
    
    func interstitialDidDismissModal(_ interstitialAd: PWAdsInterstitial!) {
        print("interstitialDidDismissModal\n")
    }
    
    func interstitialDidPresentModal(_ interstitialAd: PWAdsInterstitial!) {
        print("interstitialDidPresentModal\n")
    }
    
    func interstitialWillDismissModal(_ interstitialAd: PWAdsInterstitial!) {
        print("interstitialWillDismissModal\n")
    }
    
    func interstitial(_ interstitial: PWAdsInterstitial!, didFailWithError error: Error!) {
        print("interstitial didFailWithError\n")
        if let error = error {
            print("\(error)\n")
        }
    }
    
    func shouldLeaveApplication(for interstitial: PWAdsInterstitial!) -> Bool {
        print("shouldLeaveApplication\n")
        return true
    }
}
